packages/schematics/update/update/index.ts

```diff
@@ -33,6 +33,9 @@ type PeerVersionTransform = string | ((range: string) => string);
 // We export it to allow for testing.
 export function angularMajorCompatGuarantee(range: string) {
   range = semver.validRange(range);
```

Comment:
Can we understand why `range` is invalid in the first place? what the the inputted value?

Reply:
when running `ng update`, the range variable takes the values of all the ranges found in the package files of your dependencies. They may and may not be valid ranges for semver (the very reason .validRange is called here I suppose).

If we put that aside.  Strictly speaking, the code fail prone written as is:
```js
const output = semver.validRange(range) // output === null
...
semver.validRange(output) // fails
```